src/Rings/groebner/general.jl

```diff
@@ -142,6 +143,9 @@ function standard_basis(I::MPolyIdeal; ordering::MonomialOrdering = default_orde
     #  since msolve v0.7.0 is most of the time more efficient
     #  to compute a reduced GB by default
     groebner_basis_f4(I, complete_reduction=true)
+  elseif algorithm == :g4ti2
+    @req all(i -> length(i) == 2, gens(I)) "Ideal needs to be a lattice ideal for 4ti2 algorithm"
```

Comment:
Isn't this missing a check regarding the coefficients? As far as I can see, you only use the exponents of the gens, but never use nor check the coefficients